elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -760,6 +768,49 @@ public PersistentResource getRelation(String relation, String id) {
         return (Set) filter(ReadPermission.class, resources);
     }
 
+    @SuppressWarnings("unchecked")
+    protected Set<PersistentResource> getRelation(String relationName, final Optional<Set<Predicate>> filters,
+                                                  final Optional<Sorting> sorting,
+                                                  final Optional<Pagination> pagination) {
+        List<String> relations = dictionary.getRelationships(obj);
+
+        String realName = dictionary.getNameFromAlias(obj, relationName);
+        relationName = (realName == null) ? relationName : realName;
+
+        if (relationName == null || relations == null || !relations.contains(relationName)) {
+            throw new InvalidAttributeException(relationName, type);
+        }
+
+        Set<PersistentResource<Object>> resources = Sets.newLinkedHashSet();
+
+        // check for deny access on relationship to avoid iterating a lazy collection
+        if (isDenyFilter(requestScope, dictionary.getParameterizedType(obj, relationName))) {
+            checkFieldAwarePermissions(ReadPermission.class, relationName);
+            return (Set) resources;
+        }
+
+        RelationshipType type = getRelationshipType(relationName);
+        Object val = this.getValue(relationName);
```

Comment:
Quick question for performance. If you walk this breakpoint, you will find that all relations to the parent object are loaded into Hibernate memory. In the case of /author/{id}/books?params I saw all books by Issac Asimov get loaded prior to actually running the downstream HQL. Not sure if we are actually loading things twice, or if I was walking too deeply into the Hibernate rabbit hole.